database/database_tx.go

```diff
@@ -0,0 +1,23 @@
+package database
+
+import "database/sql"
+
+type DbTx struct {
+	*sql.Tx
+}
+
+func (db *Db) ExecuteTx(callback func(*DbTx) error) error {
+	tx, err := db.Sqlx.Begin()
+	if err != nil {
+		return err
+	}
+
+	if err := callback(&DbTx{tx}); err != nil {
+		if rollbackErr := tx.Rollback(); rollbackErr != nil {
+			return rollbackErr
```

Comment:
It would be good idea when failing to rollback to format error and return why the transaction failed in first place.